workflow/util/util.go

```diff
@@ -244,7 +246,21 @@ func SubmitWorkflow(wfIf v1alpha1.WorkflowInterface, wf *wfv1.Workflow, opts *Su
 	if err != nil {
 		return nil, err
 	}
-	return wfIf.Create(wf)
+	if opts.DryRun {
+		// Keep the workflow metadata because it will be removed by the Post request
+		workflowTypeMeta := wf.TypeMeta
```

Comment:
Here code duplication and hard to maintain. you can change the `create` function and add dry-run optional parameter.

Reply:
The [```Create```](https://github.com/argoproj/argo/blob/97e824c9a5b71baea658e8de6130bee089fb764d/pkg/client/clientset/versioned/typed/workflow/v1alpha1/workflow.go#L83) method was generated using [client-gen](https://github.com/kubernetes/code-generator/tree/master/cmd/client-gen).

I didn't want to change the method since it specifically says at the top of the file:
```
// Code generated by client-gen. DO NOT EDIT.
```

Reply:
got it